dymos/examples/ssto/launch_vehicle_ode.py

```diff
@@ -13,10 +14,153 @@ def initialize(self):
+                       units='s')
+        # Outputs
+        self.add_output('xdot',
+                        val=np.zeros(nn),
+                        desc='velocity component in x',
+                        units='m/s')
+
+        self.add_output('ydot',
+                        val=np.zeros(nn),
+                        desc='velocity component in y',
+                        units='m/s')
+
+        self.add_output('vxdot',
+                        val=np.zeros(nn),
+                        desc='x acceleration magnitude',
+                        units='m/s**2')
+
+        self.add_output('vydot',
+                        val=np.zeros(nn),
+                        desc='y acceleration magnitude',
+                        units='m/s**2')
+
+        self.add_output('mdot',
+                        val=np.zeros(nn),
+                        desc='mass rate of change',
+                        units='kg/s')
+
+        self.add_output('rho',
+                        val=np.zeros(nn),
+                        desc='density',
+                        units='kg/m**3')
+        # Setup partials
+        ar = np.arange(self.options['num_nodes'])
+
+        self.declare_partials(of='rho', wrt='y', rows=ar, cols=ar)
+        self.declare_partials(of='xdot', wrt='vx', rows=ar, cols=ar, val=1.0)
+        self.declare_partials(of='ydot', wrt='vy', rows=ar, cols=ar, val=1.0)
+
+        self.declare_partials(of='vxdot', wrt='vx', rows=ar, cols=ar)
+        self.declare_partials(of='vxdot', wrt='m', rows=ar, cols=ar)
+        self.declare_partials(of='vxdot', wrt='theta', rows=ar, cols=ar)
+        self.declare_partials(of='vxdot', wrt='y', rows=ar, cols=ar)
+        self.declare_partials(of='vxdot', wrt='thrust', rows=ar, cols=ar)
+
+        self.declare_partials(of='vydot', wrt='m', rows=ar, cols=ar)
+        self.declare_partials(of='vydot', wrt='theta', rows=ar, cols=ar)
+        self.declare_partials(of='vydot', wrt='vy', rows=ar, cols=ar)
+        self.declare_partials(of='vydot', wrt='y', rows=ar, cols=ar)
+        self.declare_partials(of='vydot', wrt='thrust', rows=ar, cols=ar)
+
+        self.declare_partials(of='mdot', wrt='thrust', rows=ar, cols=ar)
+        self.declare_partials(of='mdot', wrt='Isp', rows=ar, cols=ar)
+
+        # # Complex-step derivatives
+        # self.declare_partials(of='*', wrt='*', method='cs')
+        # self.declare_coloring(wrt='*', method='cs', show_sparsity=True)
+
+    def compute(self, inputs, outputs):
+        cb = self.options['central_body']
+
+        theta = inputs['theta']
+        cos_theta = np.cos(theta)
+        sin_theta = np.sin(theta)
+        vx = inputs['vx']
+        vy = inputs['vy']
+        m = inputs['m']
+        F_T = inputs['thrust']
+        Isp = inputs['Isp']
+        y = inputs['y']
+
+        if cb == 'earth':
+            rho_ref = 1.225
+            h_scale = 8.44E3
+        elif cb == 'moon':
+            rho_ref = 0.0
+            h_scale = 1.0
+        else:
```

Comment:
Don't need the `else` branch because self.options already does value-checking to assure that "central_body" is only in ['earth', 'moon']

Reply:
Honestly we're not even running a test against "moon" any more (and haven't been for some time).  There's no need to add this complication here, just remove `cb` as an option and replace with `rho_ref` and `h_scale` options which default to the Earth values.